docs/examples/use_cases/tensorflow/efficientdet/pipeline/dali/efficientdet_pipeline.py

```diff
@@ -0,0 +1,262 @@
+
+        self._is_training = is_training
+        self._num_shards = num_shards
+        self._shard_id = None if cpu_only else device_id
+        self._device = "cpu" if cpu_only else "gpu"
+
+        self._anchors = anchors.Anchors(
+            3, 7, 3, [1.0, 2.0, 0.5], 4.0, params["image_size"]
+        )
+        self._boxes = self._get_boxes()
+        self._max_instances_per_image = params["max_instances_per_image"] or 100
+        seed = params["seed"] or -1
+
+        self._pipe = dali.pipeline.Pipeline(
+            batch_size=self._batch_size,
+            num_threads=self._num_shards,
+            device_id=device_id,
+            seed=seed,
+        )
+        self._define_pipeline()
+
+    def _get_boxes(self):
+        boxes_t = self._anchors.boxes[:, 0] / self._image_size[0]
+        boxes_l = self._anchors.boxes[:, 1] / self._image_size[1]
+        boxes_b = self._anchors.boxes[:, 2] / self._image_size[0]
+        boxes_r = self._anchors.boxes[:, 3] / self._image_size[1]
+        boxes = tf.transpose(tf.stack([boxes_l, boxes_t, boxes_r, boxes_b]))
+        return tf.reshape(boxes, boxes.shape[0] * 4).numpy().tolist()
+
+    def _define_pipeline(self):
+        with self._pipe:
+            if self._input_type == InputType.tfrecord:
+                images, bboxes, classes, widths, heights = ops_util.input_tfrecord(
+                    self._tfrecord_files,
+                    self._tfrecord_idxs,
+                    device=self._device,
+                    shard_id=self._shard_id,
+                    num_shards=self._num_shards,
+                    random_shuffle=self._is_training,
+                )
+            elif self._input_type == InputType.coco:
+                images, bboxes, classes, widths, heights = ops_util.input_coco(
+                    self._images_path,
+                    self._annotations_path,
+                    device=self._device,
+                    shard_id=self._shard_id,
+                    num_shards=self._num_shards,
+                    random_shuffle=self._is_training,
+                )
+
+            if self._is_training and self._gridmask:
+                images = ops_util.gridmask(self._device, images, widths, heights)
+
+            images, bboxes = ops_util.normalize_flip(
+                self._device, images, bboxes, 0.5 if self._is_training else 0.0
+            )
+
+            images, bboxes, classes = ops_util.random_crop_resize(
+                self._device,
+                images,
+                bboxes,
+                classes,
+                widths,
+                heights,
+                self._image_size,
+                [0.1, 2.0] if self._is_training else None,
+            )
+
+            if self._device == "gpu":
+                bboxes = bboxes.gpu()
+                classes = classes.gpu()
+
+            enc_bboxes, enc_classes = dali.fn.box_encoder(
+                bboxes, classes, anchors=self._boxes, offset=True, device=self._device
```

Comment:
In functional API the device is inferred.
```suggestion
                bboxes, classes, anchors=self._boxes, offset=True
```

Reply:
Actually, `offset` parameter controls the mode of operation of box encoder. It either returns offsets between matches or the actual matches, so it is important regardless of whether you provide other values from the formula in the documentation.

Removed `device`.